chingu-voyages/v51-tier3-team-33
# Feature/add expense and features necessary to add an expense to a group

# Description

This is a pretty big commit, I apologize for not separating this. 
Here is a summary of all this request contains: 

Created routes for 
- Getting all users
- Getting all groups
- Adding a user to a group
- Adding an expense to a group 
- Creating a group
- Model adjustments for requirements (adding members to groups, list of expenses to groups, category to expense, etc)

## Related to the following Jira/Trello issue 
- I dont know if there is a ticket relating to adding an expense but if so, that one as well as this one below.

- As a user, I want to be able to add groups to the WeSplit app, so that I can provide a unique name to the group to enable clear understanding on what group are we splitting the expenses.

## Main changes explained:
- Created GET, POST requests for Groups
- Created GET requests for Users
- Created POST request to add existing Users to groups.
- Created POST requests for Expenses
…

## How to test:
I was unable to get automated tests set up. So the way to test all of this is use Postman and test the routes individually after starting the server. You can verify if the data works by checking the db on the mongodb website or mongdb compass if you have that setup.

## Note:
I apologize for so many changes to one branch. I had to a lot of backtracking to make sure the add expense feature worked as intended.
